jet_tracking/motorMoving.py

```diff
@@ -0,0 +1,87 @@
+"""Algorithms for Motor Moving
+
+These function definitions are created to allow for different
+mechanisms for motor moving to be tested in the various
+hutches. It is assumed for each of them that the motor
+and a dictionary of values including the range (left and right
+limits from the GUI), tolerance, and number of values to average
+for each move are included in that dictionary.
+
+This file requires that _____ be installed within the python
+environment you are running this script in
+
+this file can also be imported as a module and contains the
+following functions:
+*
+"""
+
+import logging
+from tools.motorAlgorithm import TernarySearch, BasicScan, LinearTernary, DynamicLinear
+logger = logging.getLogger(__name__)
+
+
+class MotorAction(object):
+    def __init__(self, motor_thread, context, signals):
+        self.context = context
+        self.signals = signals
+        self.motor_thread = motor_thread
+        self.ternary_search = TernarySearch(self.motor_thread, signals)
+        self.basic_scan = BasicScan(self.motor_thread, context, signals)
+        self.linear_ternary = LinearTernary(self.motor_thread, context, signals)
+        self.dyn_linear = DynamicLinear(self.motor_thread, context, signals)
+        self.motor = self.motor_thread.motor
+        self.stop_search = False
+        self.last_direction = "none"   # positive or negative or none
+        self.new_direction = "none"  # positive or negative or none
+        self.last_position = 0
+        self.new_position = 0
+        self.last_intensity = 0
+        self.new_intensity = 0
+        self.last_distance_from_image_center = 0
+        self.new_distance_from_image_center = 0
+        self.make_connections()
+
+    def make_connections(self):
+        self.signals.endEarly.connect(self.stop_the_search)
+
+    def stop_the_search(self):
+        self.stop_search = True
+
+    def execute(self):
```

Comment:
This probably could have been generalized a bit more, but it's not very crucial to do so